test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java

```diff
@@ -593,27 +593,39 @@ private void installWhiteBox() {
      * Disable IR verification completely in certain cases.
      */
     private void disableIRVerificationIfNotFeasible() {
+        if (!irVerificationPossible) {
+            return;
+        }
+
+        boolean debugTest, irTest, nonWhiteListedTest;
+
+        debugTest = Platform.isDebugBuild() && !Platform.isInt() && !Platform.isComp();
```

Comment:
I suggest to split this up into three separate checks which generate three separate messages.

Reply:
OK, fixed.